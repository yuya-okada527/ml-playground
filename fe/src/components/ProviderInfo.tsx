import React from "react";
import {
  Box,
  createStyles,
  makeStyles,
  Theme,
  Typography,
} from "@material-ui/core";
import Image from "next/image";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    providerInfo: {
      position: "absolute",
      bottom: 0,
      display: "flex",
      alignItems: "center",
    },
    providerIcon: {
      display: "inline",
    },
    providerComment: {
      fontSize: "13px",
      fontWeight: 300,
      marginRight: theme.spacing(1),
    },
  })
);

/**
 * 提供情報を描画する
 */
const ProviderInfo: React.FC = () => {
  const classes = useStyles();
  return (
    <Box className={classes.providerInfo}>
      <Typography variant="subtitle2" className={classes.providerComment}>
        This product uses the TMDb API but is not endorsed or certified by TMDb.
      </Typography>
      <Image
        src="https://www.themoviedb.org/assets/2/v4/logos/v2/blue_square_1-5bdc75aaebeb75dc7ae79426ddd9be3b2be1e342510f8202baf6bffa71d7f5c4.svg"
        alt="TMDB Logo"
        width={40}
        height={40}
        className={classes.providerIcon}
      />
    </Box>
  );
};

export default ProviderInfo;
