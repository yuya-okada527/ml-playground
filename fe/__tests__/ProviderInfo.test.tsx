import React from "react";
import { shallow } from "enzyme";
import ProviderInfo from "../src/components/ProviderInfo";

describe("ProviderInfo", () => {
  it("Snapshot Test", () => {
    const providerInfo = shallow(<ProviderInfo />);
    expect(providerInfo).toMatchSnapshot();
  });
});
