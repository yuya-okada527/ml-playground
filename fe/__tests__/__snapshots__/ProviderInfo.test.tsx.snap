// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderInfo Snapshot Test 1`] = `
<Styled(MuiBox)
  className="makeStyles-providerInfo-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-providerComment-3"
    variant="subtitle2"
  >
    This product uses the TMDb API but is not endorsed or certified by TMDb.
  </WithStyles(ForwardRef(Typography))>
  <Image
    alt="TMDB Logo"
    className="makeStyles-providerIcon-2"
    height={40}
    src="https://www.themoviedb.org/assets/2/v4/logos/v2/blue_square_1-5bdc75aaebeb75dc7ae79426ddd9be3b2be1e342510f8202baf6bffa71d7f5c4.svg"
    width={40}
  />
</Styled(MuiBox)>
`;
